api/commands/run.ts

```diff
@@ -247,10 +248,15 @@ export class Runner {
     parentAction.tasks.push(taskResult);
     await this.triggerChange();
     try {
-      const rows = await this.adapter.execute(task.statement, {
-        onCancel: handleCancel => this.eEmitter.on(CANCEL_EVENT, handleCancel),
-        maxResults: 1
-      });
+      // Retry this function a given number of times, configurable by user
+      const rows = await retryPromise(
+        () =>
+          this.adapter.execute(task.statement, {
+            onCancel: handleCancel => this.eEmitter.on(CANCEL_EVENT, handleCancel),
+            maxResults: 1
+          }),
+        this.graph.projectConfig.idempotentActionRetries || 0
```

Comment:
what about a default value of `3`?

Reply:
Scary, done

Reply:
Seems like it could make runs a lot slower to fail if a ton of stuff is failing though